k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -1652,6 +1654,104 @@ private <MSG extends Message> void updateStatus(
         }
     }
 
+    public void tryToDecryptMessagesThatCouldNotDecryptBefore() {
+        putBackground("tryToDecryptMessagesThatCouldNotDecryptBefore", null, new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    tryToDecryptMessagesThatCouldNotDecryptBeforeSynchronous();
+                } catch (MessagingException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+
+    }
+
+    private synchronized void tryToDecryptMessagesThatCouldNotDecryptBeforeSynchronous() throws MessagingException {
```

Comment:
aren't you concerned that this sync block will block execution for a while? Did we try to see if it chokes the app?

Reply:
It did not choke all the times I tried it. This is only called in very special situations as I described.
I added it because I may have found some situation during the first version of the implementation.
Nonetheless, I tried all cases removing the word synchronized and all worked find, so I removed it. Thanks for paying attention to that detail.